src/ResourceManager/KeyVault/Commands.KeyVault.Test/Scripts/VaultCertificateTests.ps1

```diff
@@ -5,14 +5,15 @@ function Get-CertificateSubjectName([string]$certificateName)
     return 'CN=' + $certificateName + '.com'
 }
 
-function Get-X509FromSecret([string]$keyVault, [string]$secretName)
+function Get-X509FromSecretBySubjectName([string]$keyVault, [string]$secretName, [string]$subjectName)
 {
     $secret = Get-AzureKeyVaultSecret $keyVault $secretName
     $secretValueBytes = [System.Convert]::FromBase64String($secret.SecretValueText)
     $x509FromSecretCollection = New-Object System.Security.Cryptography.X509Certificates.X509Certificate2Collection
     $x509FromSecretCollection.Import($secretValueBytes)
-    $x509FromSecret = $x509FromSecretCollection[0]
-    return $x509FromSecret
+    $x509Matches = $x509FromSecretCollection.Find([Security.Cryptography.X509Certificates.X509FindType]::FindBySubjectName, $subjectName, $false)
+    Assert-True { $x509Matches.Count -eq 1 }  # There should be exactly one cert in the collection with the given subject name.
```

Comment:
Can we return null here if certificate not found? Let the test assert according to context.